lib/Support/CompilationLog.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Support/CompilationLog.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/Support/FileSystem.h"
+
+#include <fstream>
+
+namespace glow {
+llvm::cl::opt<bool>
+    dumpCompilationLogOpt("dump-compilation-log",
+                     llvm::cl::desc("Dump compilation log"));
+
+std::vector<std::string> CompilationScope::scopeStack = {};
+
+CompilationScope::CompilationScope(std::string scopeName) {
+    this->scopeName = scopeName;
```

Comment:
Once you change the name of the member variable to `scopeName_`, initialize it in the constructor's initialization list:
```
CompilationScope::CompilationScope(std::string scopeName)  :  scopeName_(scopeName) {...
```

If you decide to keep the string as a member variable and a reference to it on the scope stack, you can move the provided `scopeName` into `scopeName_` instead.